ext/vscode/src/commands/deploy.ts

```diff
@@ -7,19 +7,24 @@ import { TelemetryId } from '../telemetry/telemetryId';
 import { createAzureDevCli } from '../utils/azureDevCli';
 import { executeAsTask } from '../utils/executeAsTask';
 import { isTreeViewModel, TreeViewModel } from '../utils/isTreeViewModel';
-import { AzureDevCliApplication } from '../views/workspace/AzureDevCliApplication';
+import { AzureDevCliModel } from '../views/workspace/AzureDevCliModel';
+import { AzureDevCliService } from '../views/workspace/AzureDevCliService';
 import { getAzDevTerminalTitle, getWorkingFolder } from './cmdUtil';
 
 export async function deploy(context: IActionContext, selectedItem?: vscode.Uri | TreeViewModel): Promise<void> {
-    const selectedFile = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliApplication>().context.configurationFile : selectedItem;
-
+    const selectedModel = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliModel>() : undefined;
+    const selectedFile = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliModel>().context.configurationFile : selectedItem;
     const workingFolder = await getWorkingFolder(context, selectedFile);
 
     const azureCli = await createAzureDevCli(context);
 
-    // Only supporting "deploy all" mode (no support for deploying individual services)
-    // until https://github.com/Azure/azure-dev/issues/696 is resolved.
-    const command = azureCli.commandBuilder.withArg('deploy').build();
+    const commandBuilder = azureCli.commandBuilder.withArg('deploy');
+
+    if (selectedModel instanceof AzureDevCliService) {
+        commandBuilder.withArg(selectedModel.name);
+    }
```

Comment:
I still have a remaining change to support `azd deploy --all`. `azd deploy` is sensitive to the working directory, with the impl described:
1. `azd deploy` and `azd restore` will now deploy the current service, when the current working directory is set to a service directory. 
2. `azd deploy` and `azd restore` will deploy all services, when the current working directory is set to the root project directory. 
3. In other directories, `azd deploy` and `azd restore` will not attempt a deployment and instead offer suggestions. `azd deploy --all` can be used to deploy all services, or `azd deploy <service>` to deploy a given service.

Reply:
@bwateratmsft Latest `daily` version `0.7.0-beta.2-daily.2665281` has all the changes you're looking for. Let me know if anything works unexpectedly.

Reply:
I'd agree to better be explicit.

Reply:
Ok, added the `--all` param if a service isn't specified to both `restore` and `deploy`.